Userland/Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -549,20 +549,33 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::concat)
 // 22.1.3.24 String.prototype.substring ( start, end ), https://tc39.es/ecma262/#sec-string.prototype.substring
 JS_DEFINE_NATIVE_FUNCTION(StringPrototype::substring)
 {
+    // 1. Let O be ? RequireObjectCoercible(this value).
+    // 2. Let S be ? ToString(O).
     auto string = TRY(utf16_string_from(vm));
+
+    // 3. Let len be the length of S.
     auto string_length = static_cast<double>(string.length_in_code_units());
 
+    // 4. Let intStart be ? ToIntegerOrInfinity(start).
     auto start = TRY(vm.argument(0).to_integer_or_infinity(vm));
+    // 5. If end is undefined, let intEnd be len;
     auto end = string_length;
+
     if (!vm.argument(1).is_undefined())
+        // else let intEnd be ? ToIntegerOrInfinity(end).
```

Comment:
Please don't break up spec comments across multiple lines, that kills the ability to do automated validation.

Just do this:

```
// 5. If end is undefined, let intEnd be len; else let intEnd be ? ToIntegerOrInfinity(end).
auto end = string_length;
if (!vm.argument(1).is_undefined())
    end = TRY(vm.argument(1).to_integer_or_infinity(vm));
```